### ISSUE TITLE
Build Errors

### ISSUE THREAD

**Issue by circuitcreature** [Jaffe-/NESizer2#13]
I have the synth all built and as mentioned in another issue I used the MXD1210CPA+ instead of the DS1210. I am able to flash the Atmega using an Arduino and set the fuses to  E:FF, H:DF, L:E0, or  E:FF, H:D9, L:E0.

The issue is when I power it up all the LEDs light up and it doesn't produce any sounds. I have also tried the init within the settings using buttons 8,9,11, and 14, but nothing appears to happen after pressing 11 and 14. Pressing 8 then 9 does cause the LEDs to brighten and a sequence starts on the screen for about a second.

I can however test the chip type correctly with a returned value of 12. 

Once after powering on it did start making sounds, but I haven't been able to reproduce that.

Any troubleshooting ideas?

**Comment by Jaffe-** [comment]
It is normal for it to behave strangely at first bootup, because the RAM just contains junk. The RAM is initialized properly when going to the settings mode and using 8, 9 and 11. Next boot after this it should work more as expected. 
This is one of the less finished parts of the SW, it should really just auto detect corrupt memory and initialize automatically.

If it it does not help to press, 8, 9 and 11 (when in settings mode) and power cycling, then I suspect that something might be wrong with the SRAM memories or surrounding circuitry. I would check the RAM chips (IC10, IC11) and the address latches (IC7, IC8, IC9) and the address decoder IC5 for any soldering issues. Same for the battery backup circuitry (DS1210). And of course, just to double check, you do have a battery installed?

**Comment by circuitcreature** [comment]
I put the battery in and tried the init and power cycling and everything is working. I still have the issue with the LEDs always being lit, but I can live with that.  I'll have to check my BOM, but I ended up using a different LED from the committed BOM, so there might be something in there. 
Great project thank you.

**Comment by Jaffe-** [comment]
That sounds strange. Are they always on, but brighter when actually supposed to be lit? Otherwise it sounds like it would be hard to use ...

I pushed some new code now, it checks whether the RAM has become corrupted. If it is, 'cr' will flash in the display at boot and the RAM will be properly initialized (same as manually performing it in the settings menu).

**Comment by circuitcreature** [comment]
Correct, they are alwasy on and sometimes get brighter, this is only noticeable when the init runs after pressing 8 => 9, but other than that there is no indication when on a specific mode. I used mouser #696-SSL-LX3054IID, but they look to be mostly the same as the OG BOM.

Thank you for the code update. I'll give that a try next time I get the flasher setup.

**Comment by Jaffe-** [comment]
Sounds like something might be wrong on the panel board. I would check the row and column latches there. Do all the LEDs do this, or just some of them? What about the 7 segment display?
The LEDs are very similar to the ones in the BOM, so I don't think they are the issue here.

**Comment by circuitcreature** [comment]
Double checked everything and it looks fine. It only happens to the LEDs, the display displays 00 when nothing has been selected. 
As for the update, I get a battery warning and cl on every boot and it wipes all settings.

**Comment by beau-seidon** [comment]
There are a few items in the committed BOM which are obsolete. I had to select suitable replacements. Also @Jaffe- did mention, in a reply to one of my Issues, that the BOM is not as up to date as the schematic. One example I found is that the part number listed in the BOM for the bus resistor on the top board is for a 4.9k instead of the 15k from the schematic. 

When I built mine, I opened the .sch file in Eagle (free version) and exported a new parts list, then carefully compared the committed BOM to the new list. I used both as a reference to build my own cart on Mouser, defaulting to the schematic when I found conflicts. 

I have attached an export of my Mouser cart with all the parts I used in my build, for you reference. 
Mine is working great without any electronics issues.

[exported_mouser_cart_(nesizer_only).xls](https://github.com/Jaffe-/NESizer2/files/12450320/exported_mouser_cart_.nesizer_only.xls)

**Comment by circuitcreature** [comment]
Thank you @beau-seidon I'll compare yours to mine and update as needed. I'll report back if I order any new parts.

**Comment by Jaffe-** [comment]
> Double checked everything and it looks fine. It only happens to the LEDs, the display displays 00 when nothing has been selected. As for the update, I get a battery warning and cr on every boot and it wipes all previous settings, like midi channels.

Does not sound good! I think the cr error should only occur once the first time (and yes, the settings will be erased too. I should have warned about this...) After it has written the expected signature to RAM, it should work fine after the next boot. If it does not, it means that the battery backup is not working and has corrupted the RAM.

I will check whether the battery fault is now always showing with a fresh battery also. It might need some more work to be reliable.

**Comment by beau-seidon** [comment]
> Thank you @beau-seidon I'll compare yours to mine and update as needed. I'll report back if I order any new parts.
> edit:
> Ordered more parts, the resistor array, clock, and diodes. 

If you are talking about the switching diodes, you can use the ones in the BOM (1N4148 are equivalent to the 1N914 diodes that I bought, and there will be no functional difference).


The clock will make an enormous difference if you got the wrong one. Particularly if it oscillates at a different frequency than the design. If the freq. is wrong though I wouldn't expect it to boot at all. 


Not sure what effect the resistor array difference would make, other than potentially passing too much current through the buttons if several are pressed at once? Full disclosure: I bought the wrong one too and almost said screw it I'll install it anyway lol. In the end I decided not to and got a 15k replacement before turning it on.



I'm not implying anything here, but triple check that all your IC's are oriented correctly. I booted it up the first time with my 2A03 in backwards because I am stupid, but it didn't hurt anything as far as I can tell. It still goes beep boop.

**Comment by circuitcreature** [comment]
ICs look good. I also picked up more latches, to test out. I might have messed something up on first boot, i had the pins off by one with the mainboard and panel. As far for the cystal it is 20mhz, worse case I'll build another one.

**Comment by Jaffe-** [comment]
@circuitcreature  Let me know what you find out. If you know some programming, you could try to change the update speed of the LEDs in the code. See src/task/task.c line 84 which defines the update rate for the LED refresh. If you increase this period, it should update slower and perhaps make it easier to debug. You could also try and just locally mod the code to add a mechanism to single-step the LED rows and columns from one of the buttons.

Regarding the battery level and corrupt RAM warnings, it just dawned on me that the board I currently have fully assembled is an older v0.5 board with some mods, so the battery related stuff is not the same as in the latest design. I should build a new main board based off the latest version before attempting to write any clever code, I think. I will revert the change related to battery low and memory corruption errors until I can test it on a new board.

Just out of curiosity, what do you see when you press the battery level indicator in settings? (press and hold second rightmost button on first row)
Same for you as well, @beau-seidon, what does the battery level indicator show in your case?

**Comment by circuitcreature** [comment]
@Jaffe- I've been digging around in the code, it's been a while with C. I'll look at the refresh rate.
As for the battery it displays 45. I'm pretty sure this one is new, I can dig around and see if I have another to test.

**Comment by beau-seidon** [comment]
> Just out of curiosity, what do you see when you press the battery level indicator in settings? (press and hold second rightmost button on first row) Same for you as well, @beau-seidon, what does the battery level indicator show in your case?


I haven't updated my NESizer with the most recent code you pushed. 
It is running a hex with my NoteOn&Vel=0 changes, and a hack to mute the output at boot, but no other mods.

When I check battery voltage by selecting Settings mode, and holding Button 7, I get **4.3** on my segment LED.
I can update the firmware with your most recent code and check again if you'd like.

**Comment by beau-seidon** [comment]
> Regarding the battery level and corrupt RAM warnings, it just dawned on me that the board I currently have fully assembled is an older v0.5 board with some mods, so the battery related stuff is not the same as in the latest design. I should build a new main board based off the latest version before attempting to write any clever code, I think. I will revert the change related to battery low and memory corruption errors until I can test it on a new board.

@Jaffe- I have 4 more pairs of the latest PCBs in red+blue. You want one?

**Comment by circuitcreature** [comment]
I changed the speed on the LEDS and all they do is pulse slower, but still remain on.
I've swapped out the latches on the panel board and no change with the LEDs. I also swapped the latch and the ICs next to the atmega chip and no change. Out of curiosity I removed an LED and swapped its polarity and it remained on.  

![IMG_7073](https://github.com/Jaffe-/NESizer2/assets/3423739/bda93899-cdee-48ee-b03c-2306b632c921)

**Comment by beau-seidon** [comment]
How did you orient your LEDs? 

When I built mine it wasn't exactly clear without following the traces on the PCB, which was doable, but I found that the cathode/anvil for all LEDs should face the right side of the board. 
(In the photo below they point left because the board is rotated.)

@Jaffe-  even if not the issue here, perhaps the silkscreen on the top board should be updated to show LED orientation in future versions to help prevent mistakes by new builders.


![20230906_004148.jpg](https://github.com/Jaffe-/NESizer2/assets/50247539/4ef731dd-8ace-46da-bcb3-5ef657f22ec0)

**Comment by Jaffe-** [comment]
That is right, the cathode is the flat part of the epoxy. Also before cutting the pins, the short one is the cathode on the LED. 

The silkscreen is unfortunate as @beau-seidon pointed out. It is actually indicated on the silkscreen layer if you look at panelboard_pcb.png, on the right side the circle has a flat/straight edge. However, on the produced board (at least my ones) this is not possible to see because the holes cut through it. I agree this should be more clearly indicated!

The correct orientation is that the cathode should be on the right side when viewing the board the same way as in panelboard_pcb.png.

**Comment by circuitcreature** [comment]
I had some time to work on this and got it working. The original LEDs I had were the issue. After a long slow boat shipment, I replaced them with another brand and everything works great. 

I also tested with a 2A03-G and a UA6527P, and both work great.

I've seen the recent commits and will try that version soon.

Thanks @Jaffe- and @beau-seidon !

![IMG_7280-min](https://github.com/Jaffe-/NESizer2/assets/3423739/4ba4663a-de5d-4345-8ad5-ef0e3cc0c6b7)

**Comment by beau-seidon** [comment]
Nice! I'm glad you got it working. 

It's tied with the Twisted Electrons TherapSID for my favorite chip synth.

**Comment by circuitcreature** [comment]
@Jaffe-  Not sure if you can help. I went to turn on the synth the other night and there was no sound or lights. I've checked the voltage across all the chips on the panel and several on the main board and most everything is showing 5v. 

The odd thing is that avrdude programs the chip just fine.

Any advice on things to check?

**Comment by Jaffe-** [comment]
@circuitcreature: That is weird! Have you checked the clock and its power supply too?
This might also be something that is failing during initialization. Everything will hang if the 2A03 is not working for instance. What kind of 2A03 do you have installed?

**Comment by circuitcreature** [comment]
Pin 1 on the clock is 2v and pin 2 is 2.5v
I have tried both a RP2A03-G and UA6527P

**Comment by Jaffe-** [comment]
@circuitcreature That is strange indeed. I'll try and see if I can decouple the 2A03 code a bit from the system so it can boot without it installed (probably a nice feature anyway, at least the NESizer can then be used as a MIDI sequencer :p)

In the mean time, can you try switching out the Atmega and see if that makes any difference? I did have an issue with an Atmega earlier that was somewhat like what you describe, but I thought it was just a bad chip because it worked fine when I switched it out...

**Comment by Jaffe-** [comment]
@circuitcreature I added support for running without 2A03 now. You can try building and installing that commit to see if the communication with 2A03 is the issue (although it seems unlikely as it fails with two different chips).

**Comment by beau-seidon** [comment]
try re-burning the fuse bits. I temporarily bricked my atmega chip by accidentally pressing ctrl+c in the terminal while it was flashing firmware. couldn't get it to do anything or respond to another flash command, but found that re-burning the fuse bits reset it and it's working fine again.

**Comment by circuitcreature** [comment]
Thanks @beau-seidon. I've swapped chips and finally got something, even though I think I'm back to the original chip that first started with the issues. I have it setup for debugging and it's powering on, but not functional. It could be that it's connected with test wires. I'll try again when I have some more time. I also have some more atmegas in the mail, I'll be able to test with those if all fails.

**Comment by beau-seidon** [comment]
I just got a UA6527P from eBay to test it out and it doesn't seem to work at all in the NESizer. 
My 2A03G from my NES still works perfectly. Might just be a bad chip. No idea how to troubleshoot it.

**Comment by beau-seidon** [comment]
It did not work in my NES. I contacted the seller on eBay and he refunded me. 
I'd rather have a working chip than the money though, that's why a bought it lol.

I'll test it for fun using your function when I get home, just to see what it says.

**Comment by circuitcreature** [comment]
The 6526p does work. I have a couple more i can test when I'm back up and running. I'll have some atmega chips soon. 
@beau-seidon  if this batch of 6527p works i can send you a link.

**Comment by beau-seidon** [comment]
> What do you see in the settings page if you press the 'CHIP TYPE' button (rightmost in the upper row)? If it says 0, then I would say that it is a high probability that the chip is not functioning correctly, because this means that it was not able to complete the 2A03 detection sequence (feeding it instructions and checking for R/W line transitions). 

Using the latest build, pressing the CHIP_TYPE button causes the 7 segment display to show [ - - ], when the button is held.
With my 2A03 chip in, it shows [ 1 2 ], which I expected since I removed it from my NES. 
So yeah, the UMC chip is either dead, or fake.

**Comment by beau-seidon** [comment]
This is a reply to: https://github.com/Jaffe-/NESizer2/issues/10#issuecomment-2177410174


@jeff464 please describe how your unit is "misbehaving". No lights or flickering lights? No response to button input?
Were you able to successfully set the fuse bits, and flash the firmware to the ATmega chip?  

Which software are you loading to the NESizer? 
I recommend using the @Jaffe- master branch, until you are sure it's working, then play with experimental forks/branches. 

If you have trouble compiling or uploading, try using VSCode+PlatformIO to build/load it. If you use the prebuilt .hex file, you can use avrdude (comes with ArduinoIDE) to flash it.  For more info see this link, and check out the other Closed issues: https://github.com/Jaffe-/NESizer2/issues/24#issue-2302456124

Funny noises coming from output channels?
I know the software has been updated a few times since I built mine, but this post might still be relevant, regarding first-time startup: https://github.com/Jaffe-/NESizer2/issues/7#issuecomment-926837146

Lights and input ok, but no sound at all? 
Make sure you have the voices you want to play enabled, and assigned to MIDI channels. 
Also make sure your audio cable is in the right jack and the switch is set properly if you are mixing to mono.

**Comment by jeff464** [comment]
I turn on the unit and some lights flicker/character display barely flickers.
Unit is unresponsive to any button pushes/no sound. 
All points checked out okay with the multimeter, 
Official .hex was used and wrote successfully/verified using GQ4x4 writer.

Other notes:

Command line anything isn't an option for a few more weeks - trying to maintain my sanity (build/troubleshooting now going on 90+ hours, timed)
Not entirely sure how to program fuse bits using the GQ4x4. I assume this is the root of the problem.

**Comment by jeff464** [comment]
Tried using avrdude since I'm a glutton for punishment - it opened and immediately closed despite running as administrator. Not ideal, so I found a GUI by the name of avrdudess. Cool. 

I now have an arduino uno, atmega328p, and avrdudess. However, I cannot for the life of me find instructions on how to upload the hex + fuse settings to the 328p using just the uno. Surely this is possible?

**Comment by jeff464** [comment]
Thank you for the detailed response - I sincerely appreciate it. I ended up flashing the Uno board, went out and bought another, came back, wired the two Arduino Uno boards together, and used AVRDUDESS to write the .hex w/ fuse/lock bits. Unfortunately, I am receiving the same exact behavior as before. I'm going to attempt to attack this from a few different angles over the next few hours.

**Comment by beau-seidon** [comment]
Are you programming the NESizer ATmega328p chip by installing it in an UNO/other external board?

Try putting it back in the NESizer and connecting the UNO configured as an ISP programmer directly to the header on the left side of the NESizer. 

If you are programming it while loaded in an UNO board with the 16 MHz crystal, I'm not sure how it will behave if the fuse bits are set such that the chip expects a 20 MHz clock. This was my plan originally but I never tried it, I just programmed it "In-System".

**Comment by jeff464** [comment]
I connected the side ISP pins directly to the Uno and force flashed - avrdudess wasn't recognizing the 328. I'm getting a similar result with the LEDs/display, but at least now it's responsive to a degree. When I hit the settings button, the LED lights, so that's a good sign.

That said, I'm reading about the NESizer initialization button combination in order to clear the garbage data, but I'm confused. It says to press button 8, 9, 11, and 14 while in settings mode. Is this at the same time or in order? If in order, how long do I wait between button presses? Also, those button numbers don't align with the numbers on the PCB, so I'm not sure if it actually means 9, 10, 11, and 15.

**Comment by jeff464** [comment]
This is what the display looks like when powered on.
![20240620_152014](https://github.com/Jaffe-/NESizer2/assets/63565647/ffce5f60-f9fd-40f2-bd7b-812a36c15b2b)

**Comment by beau-seidon** [comment]
It's alive! (More alive than it was anyway :) )
Press them one at a time, in the order @Jaffe- stated in that [post](https://github.com/Jaffe-/NESizer2/issues/7#issuecomment-926837146).  
The PCB silkscreen is a little out of date, and apparently so is the Settings section of the manual. This is from ui/ui_settings.c
![image](https://github.com/Jaffe-/NESizer2/assets/50247539/3961ee2d-1526-402c-8d95-778526ef666b)
![image](https://github.com/Jaffe-/NESizer2/assets/50247539/63e27f2d-3bf1-48f8-934e-95c1a693c053)

The first few go quick, but the last one takes forever because it formats the big memory chips. Give it about 5 minutes of SAMPLE_FORMAT then reboot it, and it should work!

**Comment by jeff464** [comment]
Hmmmm, I think something is up. I go to settings, hit 8, and then the four of the bottom LEDs flash rapidly. I assume this is supposed to happen, but nothing changes after about ten minutes lol. 9 seems to do something - the character display rotates briefly. 11 does nothing, and 14 also seemingly does nothing after several minutes.

If I plug in the audio cables and crank the volume, I can hear a quiet whine and tapping.

**Comment by beau-seidon** [comment]
If you reboot does it work now? Does the 7seg display 00 and change with the Up and Down buttons?
If so push one of the voice buttons (0 - 5). Does the corresponding LED stay lit? 
If so, go to Settings mode then hold MIDI Channel then select one of the voices and choose a channel, then feed it notes.

**Comment by jeff464** [comment]
It does not work after resetting, and the display does not change. Pressing channel 1 lights LED 1, 3, and 8. Pressing channel 2 lights LED 1, 2, 3, and 8. Pressing channel 3 lights LED 1,3, and 8. Pressing channel 4 lights LED 1, 2, 3, and 8. Pressing channel 5 lights LED 1,3,5, and 8. They do stay on once pressed. If I choose a a MIDI channel, LEDs flash, but I have to reset the unit for it to stop flashing. There's no audio, and I still hear clicking/whining.

**Comment by beau-seidon** [comment]
I see you have the top and bottom boards connected via jumper wires. I wonder if there's a bad connection there. 
Also when you flashed the .hex using your UNO, did you first set the fuses with the chip in the NESizer?
( Low = E0, High = D9, Extended = FF )

**Comment by jeff464** [comment]
I now cannot get any software to read (or write without force) the 328, so I'm going to assume it's bricked/corrupt.

**Comment by beau-seidon** [comment]
> After your UNO is configured and your connections are made, launch Windows command prompt and paste this command in to burn the fuse bits (replace COM4 with the COM port for your own UNO):
> 
> ```
> avrdude -C "C:\Program Files (x86)\Arduino\hardware\tools\avr\etc\avrdude.conf" -v -p atmega328p -c stk500v1 -P COM4 -b 19200 -U lfuse:w:0xe0:m -U hfuse:w:0xd9:m -U efuse:w:0xff:m
> ```
> 
> It will launch avrdude, tell the application where to find the config file, tell it to print out what it's doing, tell it what kind of chip it will be programming, what programmer it will be using, and which COM port, baud rate, and the fuse bit values to write.
> 
> You may have to re-load the .hex file to the board afterward. If so, you already know how to do that. If you want to upload it with avrdude using your UNO use this command (update the COM port and path to your .hex file before executing):
> 
> ```
> avrdude -C "C:\Program Files (x86)\Arduino\hardware\tools\avr\etc\avrdude.conf" -v -p atmega328p -c stk500v1 -P COM4 -b 1
> ```

Try this ^. Use the command line. Copy and paste those commands to a text editor, update your COM port and .hex path, connect your programmer, then paste/run those in CMD.

I oopsed my chip once by accidentally disconnecting while flashing and it wouldn't do anything anymore. Couldn't flash a new program either. I fixed it by re-burning fuse bits, and then it would let me reprogram again.

**Comment by jeff464** [comment]
I'm just getting this as a return:
`avrdude: Version 5.10, compiled on Jan 19 2010 at 10:45:23
         Copyright (c) 2000-2005 Brian Dean, http://www.bdmicro.com/
         Copyright (c) 2007-2009 Joerg Wunsch

         System wide configuration file is "C:\Program Files (x86)\Arduino\hardware\tools\avr\etc\avrdude.conf"
syntax error at C:\Program Files (x86)\Arduino\hardware\tools\avr\etc\avrdude.conf:356
`

**Comment by beau-seidon** [comment]
does that path exist and is there an avrdude.conf file in there?

I just noticed the ArduinoIDE on my work pc is not installed in "C:\Program Files (x86)".  On my laptop the path above is correct.
I found the file at the path `"%LOCALAPPDATA%\Arduino15\packages\arduino\tools\avrdude\6.3.0-arduino17\etc"` by compiling and uploading a sketch in ArduinoIDE with verbose mode on. 

You may have to update your path to the .conf file.

**Comment by jeff464** [comment]
Heh. Yeah, PlatformIO it is. The path is correct, but it's now turning into a side quest like I feared. Downloading now.

**Comment by jeff464** [comment]
Ok, I opened the nesizer.hex and this is what I'm seeing. This doesn't seem right at all.
![hex](https://github.com/Jaffe-/NESizer2/assets/63565647/3e773929-623a-440d-a163-0c0ff7f3f219)

**Comment by jeff464** [comment]
Maybe it's time I just give up on this build forever lol

**Comment by beau-seidon** [comment]
> Maybe it's time I just give up on this build forever lol

Nah, finish the build and get it working! Then give up.

> Ok, I opened the nesizer.hex and this is what I'm seeing. This doesn't seem right at all.

Yeah, that is what .hex files look like inside. It's compiled machine code. The ATmega knows what to do with it. It's not for you.

Go to the home page of this repo, then click the green `<>Code` button, and click [Download Zip](https://github.com/Jaffe-/NESizer2/archive/refs/heads/master.zip). Extract the contents and right-click on the sub-folder called NESizer2-master (with docs, hardware, src, etc. inside), then in the right-click menu choose `Open with Code` (meaning VSCode).  The PIO extension should recognize the platformio.ini file inside, and automatically download all the required AVR backend stuff. 

Open that .ini file from the Explorer panel on the left and change `upload_port = COM5` to your own COM port, and save it. 
Connect your programmer to the NEZizer, then click on the PIO icon (alien head on the left side bar). There should be a Project Tasks menu that appears at the left. Expand and click `Project Tasks / ATmega328P / Platform / Set Fuses`. This should burn the fuse bits with the configuration in the .ini file, and get the chip working again.

Next build the project using the check mark button in the top right, and you will see a "parameter name omitted" compile error :'( (I'm about to put in a pull request to fix this). 
To fix the error yourself, open `src / io / 2a03.c` and replace line 186 with `void register_set_noop(uint8_t a, uint8_t b)`. 
Then build again. Should be error-free now! (You can skip the first build step and just fix the bug first if you want...)

Next to the Build button there is a small drop down arrow, click that and select Upload. It should load the compiled firmware to your board while printing a bunch of progress messages (hopefully instead of errors) in the console.

Let me know if that helps!

**Comment by jeff464** [comment]
Thanks again for the assistance. I feel like I should be paying you. Usually I'm not this awful at DIY, I swear.

I'll follow these instructions first thing tomorrow morning. Looking forward to building this thing and then giving up. ;)

**Comment by jeff464** [comment]
First snag - there was no "open with" in the right-click menu for that folder. Obviously not a huge deal. I just opened the folder in VSC and everything seemed to populate. Cool.

Second snag - set fuses returns an error:

Selected fuses: [lfuse = 0xE0, hfuse = 0xD9, efuse = 0XFF]
Setting fuses
avrdude error: syntax error [C:\Users\Unknown:1]
avrdude error: unable to process system wide configuration file C:\Users\Unknown
*** [fuses] Error 1
=============================================================================== [FAILED] Took 3.92 seconds ===============================================================================
 *  The terminal process "C:\Users\Unknown User\.platformio\penv\Scripts\platformio.exe 'run', '--target', 'fuses', '--environment', 'ATmega328P'" terminated with exit code: 1. 
 *  Terminal will be reused by tasks, press any key to close it. 
 
 Something *REALLY* doesn't want me to build this thing.

**Comment by jeff464** [comment]
Thinking about this, it might have to do with the user folder having a space in the name. "Unknown" is only part of the username/folder, and the error seems to throw at that space. Short of editing the win registry or creating a new admin account/porting over all installed apps and programs (not a chance) I think this might officially be outside of the scope of NESizer. What a nightmare.

**Comment by circuitcreature** [comment]
It's been awhile since I've worked in the Windows CLI. You might want to try putting the paths in double quotes. e.g. "C:\User\Unknown ..." 
You can also use the tab key to auto complete paths when typing and that should auto add the quotes.

**Comment by beau-seidon** [comment]
> Thinking about this, it might have to do with the user folder having a space in the name. "Unknown" is only part of the username/folder, and the error seems to throw at that space

Yep looks like you found a bug in the python scripts that do the fuse burning and uploading. It seems to be adding extra quotes around your path which cancel out the original quotes that allow spaces in the path string. According to the PR for the issue linked below, the scripts were fixed, and I checked those installed on my system and mine reflect the PR changes. 

https://github.com/platformio/platform-atmelavr/issues/297

Apparently, it's not entirely fixed. Since it is still buggy on your system, go find that config file in 

`C:\Users\"your user name"\.platformio\packages\tool-avrdude\avrdude.conf`

and copy it to a temporary folder in `C:\` (root directory).

Then replace line 30 in platformio.ini:
`    $PROJECT_PACKAGES_DIR/tool-avrdude/avrdude.conf`
with your path to that file. Something like:
`    "C:\tmp\avrdude.conf"`

**Comment by jeff464** [comment]
Now I get a different error. This is the same error I was getting before when I realized it was no longer working. Everything is definitely hooked up properly.

Selected fuses: [lfuse = 0xE0, hfuse = 0xD9, efuse = 0XFF]
Setting fuses
avrdude: AVR device initialized and ready to accept instructions
avrdude: device signature = 0x000000 (retrying)
avrdude: device signature = 0x000000 (retrying)
avrdude: device signature = 0x000000
avrdude error: Yikes!  Invalid device signature.
avrdude error: expected signature for ATmega328P is 1E 95 0F
        Double check connections and try again, or use -F to override
        this check.


avrdude done.  Thank you.

*** [fuses] Error 1

**Comment by jeff464** [comment]
I found a post that says the error is coming from shared ISP pins, which *might* check out here considering I'm getting the same error as before when writing to the 328 when it's in-circuit with the NESizer.

https://forum.arduino.cc/t/burning-bootloader-expected-signature-for-atmega328p-is-1e-95-0f/88295/16

But really, who knows at this point,

**Comment by beau-seidon** [comment]
For the sake of thoroughness, would you mind posting a pic clearly showing the letters on top of the ATmega328 chip?

**Comment by jeff464** [comment]
Of course!
![20240621_120540](https://github.com/Jaffe-/NESizer2/assets/63565647/dcae56d2-c799-489e-81a6-b27ecab47b73)

**Comment by beau-seidon** [comment]
Are you using your Arduino UNO's ISP header or the GPIO pins 13-10 to connect to the NESizer ISP header?

Make sure use the GPIO pins (or at least Pin10 for Reset) because connecting the Reset pins of the two ISP headers together does nothing. 

Pin10 on the UNO is used as a digital output to reset the NESizer ATmega when programming it.

**Comment by jeff464** [comment]
I am currently using the ISP header from the Uno to the ISP header on the NESizer. When I was using the GPIO pins, I made sure to account for reset/pin10.

**Comment by beau-seidon** [comment]
Just to rule out interconnect problems, I recommend trying again using the GPIO pins and make sure it's connected like this:

![340854047-c001fe7f-86ff-4eac-893c-5fb60571b018](https://github.com/Jaffe-/NESizer2/assets/50247539/ced15bb3-0777-4265-b582-b4341a81734e)

This is how I connect mine and I have never had any issues, (except that one time I unplugged it while flashing the program because I'm stupid).

**Comment by jeff464** [comment]
So I connected the NESizer ISP to the GPIO pins of the Uno and got the same error. I restarted the computer and tried again, this time it recognized the 328 and seemed to write successfully. I then followed the instructions to fix line 186, successfully compiled, went to upload, and it failed. I tried the upload a second time, watched it go through the entire upload process, excitedly rebooted the NESizer, and...well...back to square one. It is behaving the same way it was before. Tried an upload two more times without VSC kicking back errors, so that's a step in the right direction.

**Comment by jeff464** [comment]
Going to go through and reflow solder and test all points a third time. Something clearly isn't right with the build.

**Comment by beau-seidon** [comment]
Something just occurred to me.... did you replace that resistor network with a 15k instead of the 4.7k?

https://github.com/Jaffe-/NESizer2/issues/10#issuecomment-2156732102

**Comment by jeff464** [comment]
Yeah, I ended up getting/installing the part listed in the updated BOM: 652-4309R-1LF-15K

**Comment by jeff464** [comment]
Well, thanks for the help, everyone. I think at this point it makes more sense to walk away for good. I re-flowed all solder and successfully tested a few points, but I'm getting the same result and can not afford the time and cost of building another. 

If anyone wants a fully built, non-working NESizer +4 top and bottom PCBs for the cost of parts, feel free to let me know.

¯\_(ツ)_/¯

**Comment by beau-seidon** [comment]
> Well, thanks for the help, everyone. I think at this point it makes more sense to walk away for good. 

Noo you're supposed to get it working BEFORE you give up. You've got the order backwards.

Another thing to try: take out the 2A03 and see how it behaves. I see your BOM says you got it on ebay. I recently got a clone on ebay that doesnt do a damn thing at all. Could be a dodgy chip causing all this grief. 

A recent update lets you boot the thing chipless so at least you can test the UI and sequencer functionality.

**Comment by jeff464** [comment]
It boots without the 2A03 in the same jacked state, so basically no difference. Tested with a second IC, and again, the same exact behavior. Bummer.

**Comment by jeff464** [comment]
Wait. I may have a had a eureka moment...

Pouring through my BOM...the character display. 

The one in the newer BOM is
https://www.mouser.com/ProductDetail/Broadcom-Avago/HDSP-523A?qs=pQfy5%252BKCabJDIZsniRXu3w%3D%3D

whereas the one in my BOM is
https://www.mouser.com/ProductDetail/696-LDD-A514RI

The biggest glaring issue is that mine is common anode, whereas the other is common cathode. I'm certain that's the issue, unless of course it's not.

**Comment by jeff464** [comment]
Though, that doesn't explain the other weirdness with the buttons. Maybe it's related?

**Comment by jeff464** [comment]
So I swapped out the display, and it looked like it flashed "E.F." on startup. When I went through the process of clearing ram, buttons 11 and 14 seemed to do nothing. I waited a while, and now it looks like it says "cr" on startup. I can confirm MIDI functionality/audio as working. Buttons 11 and 14 still don't seem to do anything during the ram clearing process, but I can at least now confirm they are functional in other modes.

So as it stands. the NESizer is 3/4 working, it displays "cr" on startup, and buttons 11 & 14 seem to have no affect on the ram clearing process.

**Comment by jeff464** [comment]
Welp, now it's saying "E.F." again after trying to reset ram a third time.

**Comment by jeff464** [comment]
Also now noticing that every power supply I'm using causes the display to read "E.F." on boot. I've tried 9v/12v, center positive/negative, even AC/DC. Nothing seems to work properly and ram clear does nothing in "EF." The only thing that returns a "cr" is a 9v battery, and the NESizer starts glitching out after a few minutes of not doing anything. The unit crashes much faster if I press up/down a few times in a row. I'm going to assume this is all related to the ram not clearing.

**Comment by jeff464** [comment]
After some experimenting, I'm seeing that all 9v batteries start out by giving the "E.F." message. However, after a minute or two, the battery drains enough for the display to read "00." At that point, if I reboot, the display will read "cr" and the unit will seemingly work, but not for very long. The NESizer does not retain patch settings when switching to a power supply and then switching back to battery.

Is there a specific power supply people here are using? I have several 9v DC center positive supplies, but they give the "E.F." messages despite being anywhere between 1000mA, and 3.3A. Also, would a faulty DS1210 cause this behavior?

**Comment by Jaffe-** [comment]
The fact that a 9V battery has to drain before working, makes it sound to me like something is wrong in the power regulation part of the NESizer. There is a 7805 regulator which is supposed to bring 9V down to 5V. So a fresh 9V battery should work fine, and actually a drained battery should fail. Can you measure the voltage coming out of the 7805? Look at this figure for reference: http://components101.com/sites/default/files/component_pin/7805-Voltage-Regulator-Pinout.png
Measure between ground and output (so pin 2 and 3). This should be 5V. To check that you use the correct DC adapter, you can also measure the input to ground voltage and see that it is as expected. You should see what is stated on the adapter, minus a small diode voltage drop (around 0.7 V).

It is also a good idea to check each solder point very thoroughly. Look for any joints that could be having too much solder and  connecting two pins together etc. The issue you have with a rapidly draining battery sounds to me like there might be a short somewhere.

**Comment by jeff464** [comment]
I will test as soon as I get back home. It's on my to-do list for this afternoon. As for the ram clearing, might you have an idea on why 11&14 seem to have no affect? I see it's also mentioned [here](https://github.com/Jaffe-/NESizer2/issues/7#issuecomment-926890148l) and [here](https://github.com/Jaffe-/NESizer2/issues/13#issue-1868395451) but there's no follow-up on how it started working for them. Also, what does "E.F." mean? Electrical failure?

**Comment by Jaffe-** [comment]
There should not be any code printing E.F. There is an error handler that can set up 'Er' in the display though, if that one happens then the top row of LEDs should indicate a fault code. Do you see any of those LEDs lighting up? What happens when you have it in this 'E.F.' state, can you press any of the buttons or is it just stuck there? 

I think the first thing to double check the power supply. The DC jack is center positive, you can also double check what you have in your BOM. If you look at the main board layout image, you can see that the two pins closest to the edge on the DC connector are grounded, and the one furthest from the edge is the positive lead. You can check in the datasheet that the component you ordered has the same pinout and use whatever type of DC adapter/supply that fits that.
As long as you use a DC supply that give 9-12 V and let's say at least 1 A, it should be OK. The one I've been using a lot is 9V and can deliver up to 1.7 A. 

What you mentioned about the display is also highly relevant, I don't see how it can work (without software change) if the polarity is swapped around (common anode vs cathode). I'm surprised that you have seen 'cr' on a display if it is the opposite polarity!

**Comment by jeff464** [comment]
Ah yes, I ended up getting a new display and swapped it out with the correct one yesterday. It turns out that the wrong display was ordered when I uploaded the original BOM to digikey and had it auto-sort/populate. I also noticed that it chose the wrong standoffs too. I went through and checked the rest of the parts, and they *should* be good, but I will triple check the regulator part number to be absolutely certain.

**Comment by beau-seidon** [comment]
> Ah yes, I ended up getting a new display and swapped it out with the correct one yesterday. It turns out that the wrong display was ordered when I uploaded the original BOM to digikey and had it auto-sort/populate.

Good to hear you got the display sorted out!

Since you're having power AND ram issues, have you double checked to make sure the coin cell battery is not upside down?

Went back to look at your BOM, but saw you removed it. I do remember seeing you got your DS1210 component on ebay rather than an updated version of the IC from a distributor like Mouser or Digikey. Was it a new part or used? A couple of the EOL chips I got from ebay have been duds. Most have been fine, but I mentioned one of the bad ones in this thread actually.

**Comment by Jaffe-** [comment]
Adding to what @beau-seidon asked, do you have a 2A03 chip installed in the NESizer? If you now for sure it is from a NES, then it's OK, but if it's something bought online, that one could also be fake and might actually cause trouble if installed.

**Comment by jeff464** [comment]
Yeah, I have a 2A03 installed - when the NESizer was working/connected to a battery I ran a check and it tested as good/legit. I was also able to test playback of each channel (except sample channel) before the NESizer crashed again.

**Comment by jeff464** [comment]
Ok, so I had a chance to test - I started with a new 9v battery and watched the power drain in real time when connected to the NESizer to the point of the unit turning off. I disconnected the battery and it tests fine at a little under 9v, so I have no idea what's happening there.

I tested a center positive, 9v DC 1.3A Triad power supply and it's showing 8.30 under load, but the 7805 is putting out 4.96. The specific regulator I'm using is [here](https://www.digikey.com/en/products/detail/stmicroelectronics/L7805CV/585964) the electrolytic caps I'm using are [here](https://www.digikey.com/en/products/detail/nichicon/UPW1H221MPD/589646) and the DC connector I'm using is [here](https://www.mouser.com/ProductDetail/151-7620E-E).

When I power on and the E.F. (Er) state happens, none of the top LEDs turn on. I can sort of navigate around, but nothing really functions.

Regarding the coin battery - text should be pointing up, correct? The part I'm using is [here](https://www.digikey.com/en/products/detail/adam-tech/BH-25F-1/13537755).

Regarding the DS1210 - I am in the process of ordering a new/updated one, since I cannot be absolutely 100% certain. It looks completely legitimate, and there haven't been any reviews claiming fake ICs, but again, better safe than sorry. That said, the NESizer is behaving the same with and without it. 

I'm not seeing/testing any shorting so far.

**Comment by jeff464** [comment]
Swapped out the DS1210 with the official/modern replacement listed in the updated BOM, and I'm still having the same issue. nothing seems to be shorting and the solder work is good. I'm at a total loss.

**Comment by beau-seidon** [comment]
Dunno dood. If you take a clear hi res picture of front and back of top and bottom boards, I'll take a look and compare to my own build and the .brd files and see if anything is different.

**Comment by jeff464** [comment]
Will do as soon as I get home. Just to confirm, people have synthesizer board 0.95 and front panel board 0.4 built and working?

**Comment by Jaffe-** [comment]
@jeff464 sorry to hear you have so much trouble getting it working. I built v0.95 last fall, although the front panel board is older. But none of the boards have been changed in years, so the others who have built it have built these versions I would think.

When you say you see E.F. and can navigate around, that sounds like things are actually working somewhat. However if it is in an error state it should show E r (just two segments lit in the 'r'). If that is the state it is in, you should be able to press the OK button (second most of the lowest two buttons on the left) to exit it. However, the only events that can cause an error state are related to MIDI, so if you don't have anything connected on MIDI it should not happen (unless there is a bug, of course ...) When you press the four center buttons under the 7 segment display, does the LED above the button follow which one you press?

One thing I remember happening on one of my builds was the Atmega simply not working. Had to replace it to get it to work. I don't know what was wrong, I just had to replace the chip. You could try that as well, although the symptoms sound a bit different here.

**Comment by jeff464** [comment]
I have no idea what happened, but I came back to it and it seems functional/responsive with a power supply that wasn't working just a few days ago. The caveat now is that the screen/LEDs are pretty messed up, so I'm doing everything blindly. I am able to adjust parameters and save, but I did notice that the Mix/Out switch doesn't have any affect on the tri/noi/dmc - they still only come out of one side. The switch works fine for the pulse channels, so I'm going to assume this is likely a soldering issue.

I'm not totally convinced the Atmega is fully working 100%, so thank you for that insight. I have another on order. Also, before I go too wild with tearing the control panel apart, I suspect the character display might be the culprit (again). Here is what I'm using - can someone confirm if I overlooked an important detail with the spec?

https://www.digikey.com/en/products/detail/broadcom-limited/HDSP-523A/2219126

**Comment by beau-seidon** [comment]
> Here is what I'm using - can someone confirm if I overlooked an important detail with the spec?

That's the same part I used, so it ought to work fine. If the synth portion is working fine, but the UI board is screwy, it may be a connectivity issue between the top and bottom. Are you still using jumper wires to connect them or did you get a longer M/M header?

**Comment by jeff464** [comment]
I got a longer header (and retested with a second pair + jumper wires again), but thinking about this further - when I was using the 9v battery, there were times where the display/functionality was totally fine...it just crashed a lot and drained power quickly. Now that it's no longer crashing/draining, the UI is all screwy. It almost like it's either/or with this thing. 

Three other things I noticed - 1) every so often I'll hear a loud "BURRRRT" totally unprovoked (which admittedly sounds pretty cool) it's almost like a digital noise and less of a thumb-on-aux-jack noise, 2) there are audible clicks in the audio path when sending MIDI notes to the NESizer, and 3) 11&14 still seemingly have no affect when in settings, but work otherwise. What should be happening during 11 & 14? LED flashing? Character animation?

**Comment by jeff464** [comment]
FWIW the LEDs for 0, 2, and 7 light up along with any other button I press on the top row while in program mode.

**Comment by beau-seidon** [comment]
It's looking a bit like you cooked one of the latches when you initially installed the other display. When you press the top row buttons, do you also see weird behavior in the display in addition to the 0 2 and 7 LEDs? Maybe swap IC1 and IC2 and see if you get an observable / repeatable change in behavior.

**Comment by beau-seidon** [comment]
> I did notice that the Mix/Out switch doesn't have any affect on the tri/noi/dmc - they still only come out of one side. The switch works fine for the pulse channels, so I'm going to assume this is likely a soldering issue.

I think your soldering is probably fine, you may just misunderstand the way the mix switch works. 
When in "MIX" position, it takes the SQ1/SQ2 output and sums it with the TRI/NOI/DMC and sends the summed signal in mono out of the jack closer to the switch. The other jack only ever outputs SQ1/SQ2.

**Comment by jeff464** [comment]
Oof. You're likely spot on with that. Some of the segments in the character display are brighter than the others, and there's no change when I swap IC1/IC2.

**Comment by beau-seidon** [comment]
> Oof. You're likely spot on with that. Some of the segments in the character display are brighter than the others, and there's no change when I swap IC1/IC2.

No change when swapped is not what I expected if only one is toast. Unless they are dead in the same way, I would have expected a different set of LED's and characters to behave badly. The latches are less than a dollar each (total is probably less than half of the shipping cost lol) so I would just replace them and see if it fixes the problem. 

> I'm not totally convinced the Atmega is fully working 100%, so thank you for that insight. I have another on order. 

Have you got the replacement Atmega chip yet? You may be able to test the one you have now by burning the Arduino Uno Bootloader to it using the ArduinoIDE and installing it into your Uno board or a breadboard with a 16 MHz clock. Then load a basic Blink sketch to it and test all the pins that connect to the top panel with an LED and resistor, change the Pin number in the sketch or loop through them. 

I doubted the MCU was bad, until you said the panel behavior did not change. 

Load a blank sketch to it before putting back in the NESizer just to protect everything, and you will have to re-set the fuse bits again for the difference in clock rate.

**Comment by jeff464** [comment]
I finally found time to come back to this - I replaced the latches with new ones, and there's no change. I will get to the new atmega in a few hours. Is it possible a diode/resistor might be causing this to happen?

**Comment by jeff464** [comment]
As-is, here's how patch 1 looks:
![1](https://github.com/user-attachments/assets/25c435ef-e042-4b12-b25e-0934fc444df3)

patch 3:
![3](https://github.com/user-attachments/assets/3dddef4d-5bdf-4fa2-ae7c-a24b1d46e53d)

patch 23:
![23](https://github.com/user-attachments/assets/0589c479-9e2c-4bb9-98da-ba4cc0ac2161)

**Comment by jeff464** [comment]
Okay, so I went back to a 9v battery and watched it drain down. When it got to a certain point, the display looked totally normal. It flashes "cr" on boot, but the battery isn't strong enough to do much before crashing. I tested two other 9v batteries that flash the weird "E.F." as mentioned before, drain down, display goes normal, then flashes the "cr" at reboot, then crashes.

I switch over to a 9v, center positive DC PSU providing 1000ma and the display is all jacked again, but functions normally.

So my theory is that resistors ain't resistin' and caps ain't cappin', specifically in the power circuit. If I starve the power (draining 9v) then it falls within tolerance, but doesn't provide enough juice. I'm not seeing any shorts, and things aren't testing oddly. Perhaps the issue is coming from the bottom board after all?

**Comment by jeff464** [comment]
Another update - I repurchased all of the parts listed in the 2nd, updated BOM, built another NESizer unit, and I'm STILL getting the same behavior. I also noticed that R3 is listed twice.

So at this point, I have two completely different units showing the same exact behavior. Either there's something wrong with the PCB manufacturing, the three PSUs I tested are all out of spec, or something changed with the board/BOM and it went untested.

Definitely at a loss.

**Comment by circuitcreature** [comment]
Im still waiting on some parts to build a new set as well, though I think my problem stems from moisture. @jeff464 can you upload some images of the complete boards in the light? It'd help others check on the parts and orientation.

**Comment by jeff464** [comment]
![20240727_175120](https://github.com/user-attachments/assets/05d330a2-dbdc-4233-8401-dc58fc0e7efa)
![20240727_175139](https://github.com/user-attachments/assets/c5dc1eb7-7000-4c0d-b39c-57e14e271776)

**Comment by jeff464** [comment]
After several days of research, a friend and I figured out that both ICs in IC5 did not work correctly despite coming directly from Mouser. I have now replaced IC5 on both units and everything seems to function properly.

That said, I'm now running into an issue with saving patches. The manual states "To save a patch, press SAVE. The button will start to blink to indicate that you can select where to store the new patch." but this isn't the case with my units. The save button does not light up or blink when pressed. Also, buttons 11 and 14 still seem to do nothing when initializing memory,

Coincidence?

**Comment by beau-seidon** [comment]
> After several days of research, a friend and I figured out that both ICs in IC5 did not work correctly despite coming directly from Mouser. I have now replaced IC5 on both units and everything seems to function properly.

NICE! sucks about the part but cool to hear it's running now! Just in time to give up, too.


Also use a bit of caution with the silkscreen on the top board; some of the button labels are out of date. Refer to manual (or source) for newest button functions.

The right button on the left pair (labeled OK) should blink when you press it, then UP / DOWN on the right side will let you choose where to store the patch. Press OK again to write to the selected storage slot.

**Comment by jeff464** [comment]
Ah. Got it. The manual still says the following: 

"To save a patch, press SAVE . The button will start to blink to indicate that you
can select where to store the new patch. Use UP and DOWN to select where to store the patch. Press SAVE again to store the patch at the selected location.To save a patch, press SAVE . The button will start to blink to indicate that you can select where to store the new patch. Use UP and DOWN to select where to store the patch. Press SAVE again to store the patch at the selected location."

Thanks for the clarification. I'm hoping to test the alternate firmware against famimimidi this week.

**Comment by Jaffe-** [comment]
@jeff464: Happy you got it working. Congratulations with a successful build! Really strange that both ICs failed and came straight from Mouser. Perhaps from a bad batch?

**Comment by circuitcreature** [comment]
Follow up on my issue. I was able to track it down to a bad SRAM chip. not sure what happened to it, but after swapping it out the synth booted up just fine.

**Comment by circuitcreature** [comment]
Anyone see any reason to not close this issue, the original problem has solved.
